server/logger/logger.go

```diff
@@ -0,0 +1,62 @@
+package logger
+
+import (
+	"log"
+	"os"
+	"io/ioutil"
+	"io"
+)
+
+const loggerFlags = log.Ldate | log.Ltime
+
+
+type Logger struct{
+	debugLogger log.Logger
+	infoLogger log.Logger
+	warningLogger log.Logger
+	errorLogger log.Logger
+	fatalLogger log.Logger
+}
+
+
+func InitializeLogger(level uint8) *Logger {
+	return &Logger{
+        debugLogger:   *log.New(AssignWriter(1, level), "", loggerFlags),
+        infoLogger:    *log.New(AssignWriter(2, level), "", loggerFlags),
+        warningLogger: *log.New(AssignWriter(3, level), "", loggerFlags),
+        errorLogger:   *log.New(AssignWriter(4, level), "", loggerFlags),
+        fatalLogger:   *log.New(AssignWriter(5, level), "", loggerFlags),
+    }
+}
+
+func AssignWriter(level uint8, givenLevel uint8) io.Writer{
+	if givenLevel<level{
```

Comment:
Ten warunek odwrotnie. Chcielibyśmy, aby najbardziej szczegółowym poziomem był debug. Jeżeli debug będzie, to wszystkie inne też. Jeżeli będzie info, to warning, error i fatal też. Jeżeli będzie warning, to error i fatal też. I tak dalej